.teamcity/_self/lib/wpcom/WPComPluginBuild.kt

```diff
@@ -175,6 +175,11 @@ open class WPComPluginBuild(
 					# Note: we exclude asset changes because we only really care if the build files (JS/CSS) change. That file is basically just metadata.
 					if [ "%skip_release_diff%" = "true" ] || ! diff -rq --exclude="*.asset.php" --exclude="build_meta.json" --exclude="README.md" $archiveDir ./release-archive/ ; then
```

Comment:
@noahtallen Not directly related to this changeset, but I'm wondering about the `if [ "%skip_release_diff%" = "true" ]` condition here. Shouldn't it be `!=` so that it will actually skip if it's falsey? As far as I understand, now it _will_ diff if this var is `true`, which seems wrong as the name is `skip_release_diff`. Or am I missing something?

Reply:
I'll admit it's definitely confusing. But the goal of that flag is to publish a new release _regardless_ of whether anything changed. So if that flag is true the following happens:

1. The "old" release archive is not downloaded
2. No comparison is run between the new and old archives (e.g. the `diff` command never executes)
3. A release is published anyways

This exists so that if an older release doesn't exist for some reason (e.g. the app is new) or we need to force-publish something, then we can publish an archive anyways with a simple checkbox.

So the `skip_release_diff` is very literal: we're skipping the `diff` against the previous release which determines if anything changed.